### PR TITLE
MM-37922 - Add support for Chimera to GitLab plugin

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary

Chimera Proxy is an easier way to connect plugins to external OAuth providers. It provides pre-registered OAuth applications allowing users to skip the app registration step.

Design doc: https://docs.google.com/document/d/1a1AzNwxrqA5bcVonZOVu94b4vV6tT4U7fDIKBwf_aUI/edit

This PR makes it possible to connect GitLab Plugin via Chimera. It introduces:
- New config option, whether to use pre-registered application (Chimera)
- Changes validation if connection through Chimera is enabled
- Creates different OAuth config when using Chimera

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37922

### CODE REVIEW

**Review by iomodo — APPROVED** [review]
LGTM! Thanks @Szymongib

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
 dist
+
+# IDEs
+/.idea
+/.vscode
```
Comment (hanzei) [inline .gitignore:5]: Nit pick: Unrelated change

**Review by hanzei — APPROVED** [review]
Great work :+1:

**Comment by hanzei** [comment]
@DHaussermann See https://github.com/mattermost/mattermost-plugin-github/pull/467#issuecomment-890998338 of for the tests steps.

**Comment by jasonblais** [comment]
@DHaussermann Kind reminder to help with QA review. Similar to https://github.com/mattermost/mattermost-plugin-github/pull/467, we'd like to get ready to ship the changes ahead of the Oct 13 launch, and would be happy to help where we can.

As an aside, I wonder if Demansol would be equipped to help test the changes here or at https://github.com/mattermost/mattermost-plugin-github/pull/467?

**Comment by Szymongib** [comment]
To align with the implementation of the GitHub plugin and support both v5 and v6 servers I have added a fallback to Chimera env var. For more context see: https://github.com/mattermost/mattermost-plugin-github/pull/467#issuecomment-918178108

**Comment by Szymongib** [comment]
@DHaussermann I have aligned the setting text with the GitHub plugin.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Functional testing completed
- I know see the text change on config as expected

Thanks @Szymongib 

LGTM!
